pkg/client/unversioned/clientcmd/api/types.go

```diff
@@ -140,3 +144,9 @@ func NewAuthInfo() *AuthInfo {
 func NewPreferences() *Preferences {
 	return &Preferences{Extensions: make(map[string]runtime.Object)}
 }
+
+// AuthProviderConfig
+type AuthProviderConfig struct {
+	Name   string           `json:"name"`
+	Config *json.RawMessage `json:"config"`
```

Comment:
I noticed that there is already a `runtime.RawExtension`, in the v1 clientcmd types, and a corresponding `runtime.Object` in the unversioned types. Can I just reuse that? Or am I better off defining a new field?

I'm still looking for where (if) the extensions are actually registered/used...